webui/src/pages/auth/groups/index.tsx

```diff
@@ -149,23 +152,26 @@ const GroupsContainer = () => {
 
             <DataTable
                 results={results}
-                headers={['', 'Group ID', 'Permission', 'Created At']}
+                headers={headers}
                 keyFn={group => group.id}
-                rowFn={group => [
-                    <Checkbox
-                        name={group.id}
-                        onAdd={() => setSelected([...selected, group])}
-                        onRemove={() => setSelected(selected.filter(g => g !== group))}
-                    />,
-                    <Link href={{pathname: '/auth/groups/:groupId', params: {groupId: group.id}}}>
-                        {group.id}
-                    </Link>,
-                    group.acl ? <ACLPermission initialValue={group.acl.permission} onSelect={
-                        ((permission) => auth.putACL(group.id, {...group.acl, permission})
-                            .then(() => setPutACLError(null), (e) => setPutACLError(e)))
-                    }/> : <></>,
-                    <FormattedDate dateValue={group.creation_date}/>,
-            ]}/>
+                rowFn={group => {
+                    const elements = [
+                        <Checkbox
+                            name={group.id}
+                            onAdd={() => setSelected([...selected, group])}
+                            onRemove={() => setSelected(selected.filter(g => g !== group))}
+                        />,
+                        <Link href={{pathname: '/auth/groups/:groupId', params: {groupId: group.id}}}>
+                            {group.id}
+                        </Link>]
+                    rbac === 'simplified' && elements.push(group.acl ? <ACLPermission initialValue={group.acl.permission} onSelect={
+                            ((permission) => auth.putACL(group.id, {...group.acl, permission})
+                                .then(() => setPutACLError(null), (e) => setPutACLError(e)))
+                        }/> : <></>)
+                    elements.push(<FormattedDate dateValue={group.creation_date}/>)
```

Comment:
This is a bit harder for me to grasp actually, because it modifies an Array.  Personally I prefer the previous style.

Reply:
So do I...
The problem I had with the previous style and making the Permissions table optional is when inserting null inside the array the column is rendered as an empty column.